src/sql/workbench/contrib/tableDesigner/browser/actions.ts

```diff
@@ -42,3 +35,28 @@ export class SaveTableChangesAction extends Action {
 		this._onStateChangeDisposable?.dispose();
 	}
 }
+
+export class SaveTableChangesAction extends SaveTableChangesActionBase {
+	public static ID = 'tableDesigner.saveTableChanges';
+	public static LABEL = localize('tableDesigner.saveTableChanges', "Save Changes...");
+	constructor() {
+		super(SaveTableChangesAction.ID, SaveTableChangesAction.LABEL, Codicon.save.classNames);
```

Comment:
1. Does "generating a report" also result in the designer being saved? That doesn't seem like something that is clear by name alone. I would suggest making it explicit that generating a report will result in the designer state being saved as well if that's necessary (and if it isn't then either add an option to generating a report to tell it to save first or just simply call save then call generate report after). To me it makes more sense to split them out (what if a user wants to save their progress but doesn't care about seeing the report?)

1. The whole concept of saving seems a bit strange - it's not actually persisting the change data locally anywhere, right? What is your description of what should happen when a user "saves" the editor? It seems like if we have a save it should just be saving it locally (if even doing that) - and then for creating/updating the table we should have a `publish` action instead or something (similar to DacFX)

1. And then along with that - using the "save" icon is definitely confusing in my eyes if nothing is actually being saved anywhere. (such as when they generate a report instead). Would chat with designers about this one...an up arrow or something (for publishing) seems clearer to me. 

1. It seems a little strange to have the "save" button also then ask to generate the script when there's a button to generate the script already on the designer toolbar. Is having that in both places actually necessary?

Reply:
a report here means a summary of the changes that will be applied to the database, I will rename it to avoid confusion. also I will change the 'save' to 'publish' and find an icon for publish.

regarding to whether it is necessary to show the "generate script" twice, I think it is not a huge deal, I can remove it from the dialog later if people don't like it.